geniza/corpus/admin.py

```diff
@@ -155,6 +155,25 @@ def clean(self):
             )
 
 
+class HasTranscriptionListFilter(admin.SimpleListFilter):
+    """Custom list filter for documents with associated transcription content"""
+
+    title = "Transcription"
+    parameter_name = "transcription"
+
+    def lookups(self, request, model_admin):
+        return (
+            ("yes", "Has transcription"),
+            ("no", "No transcription"),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == "yes":
+            return queryset.filter(footnotes__content__html__regex=r".*").distinct()
+        if self.value() == "no":
+            return queryset.exclude(footnotes__content__html__regex=r".*").distinct()
```

Comment:
This look good. Wondering about performance of `__regex=r".*"` vs `__exact=""`—likely insignificant but thought I'd mention. Though I've now read online that this regex should match an empty string, so would potentially have a different result from `__exact=""`. Is that true?

Reply:
Thanks for pushing me on this, agree that regex should generally be a last resort option. It turns out there is a `has_key` filter for json fields!

And sync transcription script doesn't generate content with empty html (it checks for and skips empty TEI files).

Reply:
Perfect! :+1